external-crates/move/crates/move-compiler/src/cfgir/optimize/mod.rs

```diff
@@ -46,12 +49,17 @@ pub fn optimize(
     env: &mut CompilationEnv,
     package: Option<Symbol>,
     signature: &FunctionSignature,
-    locals: &UniqueMap<Var, (Mutability, SingleType)>,
+    infinite_loop_starts: &BTreeSet<Label>,
+    locals: &mut UniqueMap<Var, (Mutability, SingleType)>,
     constants: &UniqueMap<ConstantName, Value>,
     cfg: &mut MutForwardCFG,
+    is_constant: bool,
```

Comment:
A question about naming: why `is_constant` here? 
It seems to be doing that and a bit more? Since it is controlling the loop for coalesce_locals